HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -127,6 +127,47 @@ def customizeHLTfor47079(process):
                     
     return process
 
+
+def customizeHLTfor47047(process):
+    """Migrates many ESProducers to MoveToDeviceCache"""
+    import copy
+    if hasattr(process, "ecalMultifitParametersSource"):
+        del process.ecalMultifitParametersSource
+    esProducer = None
+    for prod in esproducers_by_type(process, "EcalMultifitParametersHostESProducer@alpaka"):
+        if esProducer is not None:
+            raise Exception("Assumption of only one EcalMultifitParametersHostESProducer@alpaka in a process broken")
+        esProducer = prod
+    if esProducer is not None:
+        for prod in producers_by_type(process, "EcalUncalibRecHitProducerPortable@alpaka", "alpaka_serial_sync::EcalUncalibRecHitProducerPortable"):
+            for attr in ["EBtimeFitParameters", "EEtimeFitParameters", "EBamplitudeFitParameters", "EEamplitudeFitParameters"]:
+                setattr(prod, attr, copy.deepcopy(getattr(esProducer, attr)))
+        delattr(process, esProducer.label())
+
+    for prod in producers_by_type(process, "HBHERecHitProducerPortable@alpaka", "alpaka_serial_sync::HBHERecHitProducerPortable"):
+        pulseOffsetLabel = prod.mahiPulseOffSets.getModuleLabel()
```

Comment:
I realize only now, this this could have been written better. If one tries to run this customization function on a menu that has already the customization re-absorbed (e.g. `/dev/CMSSW_15_0_0/GRun`) one incurs in runtime errors such as: `'EDProducer' object has no attribute 'mahiPulseOffSets'`.

@Sam-Harper FYI

Reply:
> If one tries to run this customization function on a menu that has already the customization re-absorbed

Is this situation something these customize functions should generally work? I'm sorry I didn't think about that (I though the customize function would be discarded after the menu itself has been updated).

Reply:
> Is this situation something these customize functions should generally work?

when we are transitioning between release cycle, yes, it happens commonly.

Reply:
This has been addressed at https://github.com/cms-sw/cmssw/pull/47393 and backported at https://github.com/cms-sw/cmssw/pull/47395